src/app/common/theme-provider.tsx

```diff
@@ -0,0 +1,64 @@
+import {
+  createContext,
+  Dispatch,
+  FC,
+  SetStateAction,
+  useContext,
+  useEffect,
+  useState,
+} from 'react';
+
+type ComputedTheme = 'Dark' | 'Light';
+export type UserSelectedTheme = ComputedTheme | 'System';
+
+const ThemeContext = createContext<
+  [ComputedTheme, UserSelectedTheme, Dispatch<SetStateAction<UserSelectedTheme>>]
+>([
+  // These values are not used, but are set to satisfy the context's value type.
+  'Light',
+  'System',
+  () => {},
```

Comment:
Providing initial values is required due to the context's type not accounting for `null`/`undefined`, which if it did, would make the context value quite tedious to use: checking for these values would be necessary each time the context is accessed.